osquery/extensions/tests/extensions_tests.cpp

```diff
@@ -31,20 +31,50 @@ namespace osquery {
 const int kDelayUS = 2000;
 const int kTimeoutUS = 1000000;
 
+#ifdef WIN32
+static bool namedPipeExists(const std::string& path) {
+  // Check to see if NamedPipe is alive, time out after 10ms
+  if (::WaitNamedPipeA(path.c_str(), 10) == 0) {
+    DWORD error = ::GetLastError();
+    if (error == ERROR_FILE_NOT_FOUND) {
+      return false;
+    } else if (error == ERROR_BAD_PATHNAME) {
+      return false;
+    }
+  }
+
+  return true;
+}
+#endif
+
 class ExtensionsTest : public testing::Test {
  protected:
   void SetUp() {
-    socket_path = kTestWorkingDirectory + "test.em" + std::to_string(rand());
+#ifdef WIN32
+    socket_path = "\\\\.\\pipe\\";
+#else
+    socket_path = kTestWorkingDirectory;
+#endif
+
+    socket_path += "testextmgr" + std::to_string(rand());
+
+#ifdef WIN32
+    if (namedPipeExists(socket_path)) {
```

Comment:
The scattered logic for using `namedPipeExists` and `pathExists` (for a domain socket) is becoming weird, we should collapse these two checks into 1.